### PR TITLE
memory_dff: Fix checking of feedback mux input when more than one mux

### PR DESCRIPTION
**Pull request by daveshah1**
Fixes a regression from #1130

cc @gsomlo, this was found in https://github.com/gsomlo/yoloRISC

### CODE REVIEW

**Comment by daveshah1** [comment]
Assuming we are happy with this, I think this should be added to the 0.9 RC as it fixes a relatively serious regression. Not sure how we want to approach that (either cherry-pick, or submit this PR straight to yosys-0.9-rc instead?)

**Comment by gsomlo** [comment]
Tested (with and without "-nowidelut -abc9") on both yoloRISC and LiteX+RocketChip (https://github.com/enjoy-digital/litex); fixes regression on both -- thanks for the quick turnaround!

**Comment by eddiehung** [comment]
Looks good. Sorry this was something I introduced.
The only minor concern I have is that when it comes to support reset-able read ports, this needs to be revisited.

File: passes/memory/memory_dff.cc [hunk 1]
```diff
@@ -197,7 +198,8 @@ struct MemoryDffWorker
 				if (sigbit_users_count[bit] > 1)
 					goto skip_ff_after_read_merging;
 
-			if (find_sig_before_dff(sig_data, clk_data, clk_polarity, true) && clk_data != RTLIL::SigSpec(RTLIL::State::Sx) && sig_data == check_q)
+			if (find_sig_before_dff(sig_data, clk_data, clk_polarity, true) && clk_data != RTLIL::SigSpec(RTLIL::State::Sx) &&
+					std::all_of(check_q.begin(), check_q.end(), [&](const SigSpec &cq) {return cq == sig_data; }))
```
Comment (eddiehung) [inline passes/memory/memory_dff.cc]: Wait, this checks that all entries in `check_q` are equal to `sig_data`?

Comment (daveshah1) [reply passes/memory/memory_dff.cc]: Yes, at this point `sig_data` is actually the FF output because of `find_sig_before_dff`'s behaviour

Comment (eddiehung) [reply passes/memory/memory_dff.cc]: Ahh never mind, I see what it's doing. Merge when ready.

**Pull request merged**